src/bin/bat/main.rs

```diff
@@ -86,6 +86,19 @@ pub fn list_languages(config: &Config) -> Result<()> {
         .filter(|syntax| !syntax.hidden && !syntax.file_extensions.is_empty())
         .cloned()
         .collect::<Vec<_>>();
+
+    for lang in languages.iter_mut() {
+        let lang_name = lang.name.clone();
+        lang.file_extensions.retain(|extension| {
+            let test_file = Path::new("test").with_extension(extension);
+            match config.syntax_mapping.get_syntax_for(test_file) {
+                Some(MappingTarget::MapTo(primary_lang)) => lang_name == primary_lang,
+                Some(MappingTarget::MapToUnknown) => true,
```

Comment:
Shouldn't this be `false`?

Reply:
I think you're right, it should be `false`.